File: Entrega final/Java/NoLineales/Secant.java
package me.DavidLake.AnalisisNumerico.NoLineales;

import me.DavidLake.AnalisisNumerico.Main;

public class Secant {

    private static double f(double x){

        return Main.f(x);
    }

    public static void solve(double x0, double x1, double tolerancia){

        if(x1 <= x0){
            System.out.println("Intervalo no válido.");
            return;
        }

        double fx0 = f(x0);
        int actualIterations = 0;
        double absoluteError = Double.MAX_VALUE;
        double relativeError;

        System.out.printf("\n%4s %15s %23s %22s %27s\n\n", "iter", "xn", "f(x)", "ErrorAbs", "ErrorRelativo");

        if (fx0 == 0) {

            System.out.printf("\nX0 = %f es una Raíz\n", x0);
        } else {

            double fx1 = f(x1);
            double denominator = fx1 - fx0;
            actualIterations++;

            System.out.printf("%4d %+1.20f %+1.20f %10s %10s\n", actualIterations, x0, fx0, "", "");
            System.out.printf("%4d %+1.20f %+1.20f %+1.20f %+1.20f\n", actualIterations, x1, fx1, (x1-x0), (x1-x0)/x1); //Error " "

            while (absoluteError > tolerancia && fx1 != 0 && denominator != 0 && actualIterations < Main.maxIter) {

                double x2 = x1 - ((fx1 * (x1 - x0)) / denominator);
                absoluteError = Math.abs(x2 - x1);
                relativeError = absoluteError / Math.abs(x2);

                x0 = x1;
                fx0 = fx1;
                x1 = x2;
                fx1 = f(x1);

                denominator = fx1 - fx0;
                actualIterations++;

                System.out.printf("%4d %+1.20f %+1.20f %+1.20f %+1.20f\n", actualIterations, x1, fx1, absoluteError, relativeError);
            }
        }

        if(actualIterations >= Main.maxIter && absoluteError > tolerancia) System.out.println("\nNo se llegó a un resultado en "+actualIterations+" iteraciones.");
        else System.out.printf("\nLa raiz es: %.15f\n", x1);
    }
}

File: Entrega final/webSite/contenido/metodos/oneVariable/secant.py
from math import *


# err_type = abs | rel | fx
def secant(f, interval, tol, n, err_type):
    x0 = interval[0]
    x1 = interval[1]
    assert x1 > x0
    table = []

    fx0 = f(x0)
    actual_iterations = 0

    if fx0 == 0:
        return x0
    else:
        error = float("inf")

        fx1 = f(x1)

        table.append(actual_iterations)
        table.append(x0)
        table.append(x1)
        table.append(fx0)
        table.append(fx1)
        table.append("")
        table.append("")
        table.append("newline")

        if err_type == "fx":
            error = abs(fx1)

        denominator = fx1 - fx0
        actual_iterations = actual_iterations + 1
        while error > tol and fx1 != 0 and denominator != 0 and actual_iterations < n:
            x2 = x1 - ((fx1 * (x1 - x0)) / denominator)
            abs_err = abs(x2 - x1)
            rel_err = abs_err / abs(x2)

            if err_type == "rel":
                error = rel_err
            else:
                error = abs_err

            x0 = x1
            fx0 = fx1
            x1 = x2
            fx1 = f(x1)

            if err_type == "fx":
                error = abs(fx1)

            denominator = fx1 - fx0
            actual_iterations = actual_iterations + 1

            table.append(actual_iterations)
            table.append(x0)
            table.append(x1)
            table.append(fx0)
            table.append(fx1)
            table.append(abs_err)
            table.append(rel_err)
            table.append("newline")

    if fx1 == 0:
        table.append("nan")
        return x1, table
    elif error < tol:
        return x1, table
    else:
        table.append("nan")
        return None, table
